src/components/data/listing/TableView.js

```diff
@@ -254,182 +280,222 @@ function TableView(props) {
+                                                        detailsEnabled
+                                                    }
+                                                    localUploadId={
+                                                        localUploadId
+                                                    }
+                                                    MenuProps={{ tabIndex: 0 }}
+                                                    onDownloadSelected={() =>
+                                                        onDownloadSelected(
+                                                            resourceId
+                                                        )
+                                                    }
+                                                    onEditSelected={() =>
+                                                        onEditSelected(
+                                                            resourceId
+                                                        )
+                                                    }
+                                                    onMetadataSelected={() =>
+                                                        onMetadataSelected(
+                                                            resourceId
+                                                        )
+                                                    }
+                                                    onDeleteSelected={() =>
+                                                        onDeleteSelected(
+                                                            resourceId
+                                                        )
+                                                    }
+                                                    setUploadDialogOpen={
+                                                        setUploadDialogOpen
+                                                    }
+                                                    setImportDialogOpen={
+                                                        setImportDialogOpen
+                                                    }
+                                                    selected={selected}
+                                                    uploadMenuId={uploadMenuId}
+                                                />
+                                            </TableCell>
+                                        </TableRow>
+                                    );
+                                })}
+                        </TableBody>
+                    )}
+                </Table>
+            </TableContainer>
+            <FileBrowser
+                id={localUploadId}
+                handleUploadFiles={handleUploadFiles}
+            />
+            <UploadDialog
+                open={uploadDialogOpen}
+                handleClose={() => setUploadDialogOpen(false)}
+            />
+            <URLImportDialog
```

Comment:
If I'm understanding right, we've basically moved up the responsibility for these upload and import dialogs here, which is good.  But I think that also means they can be removed from the render functions in wherever else they were originally (I think `Toolbar` and maybe also `UploadMenuBtn`)?  I may have missed something.

Reply:
Hmm. Honestly that didn't cross my mind. Although when I Look at the code, I still think it is needed at both the places. `Toolbar` has its own dot menu and table view has its own dot menu.

Reply:
Yeah, I don't see a reference in here to the toolbar, however the `Listing` component uses both the toolbar and this `TableView`. So maybe it's possible to pull up only 1 definition of each of these dialogs and the browser into the `Listing` component, and then the toolbar and this table view can make callbacks up to the `Listing` component to open the dialogs?